src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc

```diff
@@ -0,0 +1,351 @@
+  return it != pending_requests_.end();
+}
+
+bool RCPendingResumptionHandler::IsAnotherAppsSubscribedOnTheSameModule(
+    const uint32_t app_id, const ModuleUid& module_uid) const {
+  auto get_subscriptions = [](application_manager::ApplicationSharedPtr app) {
+    std::set<ModuleUid> result;
+    auto rc_app_extension = RCHelpers::GetRCExtension(*app);
+    if (rc_app_extension) {
+      result = rc_app_extension->InteriorVehicleDataSubscriptions();
+    }
+    return result;
+  };
+
+  auto app_subscribed = [module_uid, &get_subscriptions](
+                            application_manager::ApplicationSharedPtr app) {
+    auto subscriptions = get_subscriptions(app);
+    auto it = subscriptions.find(module_uid);
+    return subscriptions.end() != it;
+  };
+
+  std::set<uint32_t> subscribed_apps;
+  for (auto& app : application_manager_.applications().GetData()) {
+    if (app_subscribed(app)) {
+      LOG4CXX_DEBUG(logger_,
+                    "APP " << app->app_id() << " subscribed "
+                           << module_uid.first << " " << module_uid.second);
+      subscribed_apps.insert(app->app_id());
+    }
+  }
+
+  subscribed_apps.erase(app_id);
+  return !subscribed_apps.empty();
+}
+
+utils::Optional<smart_objects::SmartObject>
+RCPendingResumptionHandler::GetPendingRequest(const uint32_t corr_id) {
+  auto corr_id_match = [corr_id](const PendingRequest& item) {
+    return corr_id == item.correlation_id();
+  };
+  auto it = std::find_if(
+      pending_requests_.begin(), pending_requests_.end(), corr_id_match);
+  if (it == pending_requests_.end()) {
+    return utils::Optional<smart_objects::SmartObject>::EMPTY;
+  }
+  return it->message;
+}
+
+utils::Optional<uint32_t> RCPendingResumptionHandler::GetPendingApp(
+    const uint32_t corr_id) {
+  auto corr_id_match = [corr_id](const PendingRequest& item) {
+    return corr_id == item.correlation_id();
+  };
+  auto it = std::find_if(
+      pending_requests_.begin(), pending_requests_.end(), corr_id_match);
+  if (it == pending_requests_.end()) {
+    return utils::Optional<uint32_t>::EMPTY;
+  }
+  return it->app_id;
+}
+
+void RCPendingResumptionHandler::RemovePendingRequest(const uint32_t corr_id) {
+  auto corr_id_match = [corr_id](const PendingRequest& item) {
+    return corr_id == item.correlation_id();
+  };
+  auto it = std::find_if(
+      pending_requests_.begin(), pending_requests_.end(), corr_id_match);
+  if (it != pending_requests_.end()) {
+    pending_requests_.erase(it);
+  }
+  LOG4CXX_WARN(logger_,
+               "Pending request with corr_id " << corr_id << " not found");
+}
+
+void RCPendingResumptionHandler::AddPendingRequest(
+    const uint32_t app_id, const smart_objects::SmartObject request_so) {
+  namespace am_strings = app_mngr::strings;
+  auto cid = request_so[am_strings::params][am_strings::correlation_id].asInt();
+
+  pending_requests_.push_back({app_id, request_so});
+}
```

Comment:
```suggestion
void RCPendingResumptionHandler::AddPendingRequest(
    const uint32_t app_id, const smart_objects::SmartObject request_so) {
  pending_requests_.push_back({app_id, request_so});
}
```
`cid` is unused

Reply:
@jacobkeeler 
Fixed in:https://github.com/smartdevicelink/sdl_core/commit/b0245f2b264491e376b7f23d915ef6534a2492ca